### ISSUE TITLE
The code for the BEIT2 segmentation task contains bugs.

### ISSUE THREAD

**Issue by jsrdcht** [microsoft/unilm#1061]
**Describe the bug**
Model I am using (UniLM, MiniLM, LayoutLM ...): BEIT2

The problem arises when using:
* [ ] the official example scripts: (give details below)

1.  The default configuration file **upernet_beit_base_12_512_slide_160k_ade20k.py** contains a redundant parameter _rel_pos_bias_interpolation_type=0_ that is not implemented at all.
2. The version of mmseg used in the code is too old, and now the new environment requires modification of the code for loading pre-trained models in order to start. This makes it impossible to verify the code library seamlessly.
3. After making the minimum code modifications, I successfully ran the script on mmseg=0.30.0. However, the default parameters did not yield the claimed accuracy, and even fell significantly below it. My modified code is shown below.

**To Reproduce**
1. 
```python
@BACKBONES.register_module()
class MSBEiT(BaseModule):
    """ Vision Transformer with support for patch or hybrid CNN input stage
    """
    def __init__(self, img_size=224, patch_size=16, in_chans=3, num_classes=80, embed_dim=768, depth=12,
                 num_heads=12, mlp_ratio=4., qkv_bias=False, qk_scale=None, drop_rate=0., attn_drop_rate=0.,
                 drop_path_rate=0., hybrid_backbone=None, norm_layer=None, init_values=None, use_checkpoint=False, 
                 use_abs_pos_emb=True, use_rel_pos_bias=False, use_shared_rel_pos_bias=False,
                 out_indices=[3, 5, 7, 11],pretrained=None,init_cfg=None):
        super().__init__(init_cfg=init_cfg)

        assert not (init_cfg and pretrained), \
            'init_cfg and pretrained cannot be set at the same time'
        
        if isinstance(pretrained, str):
            warnings.warn('DeprecationWarning: pretrained is deprecated, '
                          'please use "init_cfg" instead')
            print('DeprecationWarning: pretrained is deprecated, ')
            self.init_cfg = dict(type='Pretrained', checkpoint=pretrained)
        elif pretrained is not None:
            raise TypeError('pretrained must be a str or None')


        ###
        the same code
        ###


        self.init_weights()

    def fix_init_weight(self):
        def rescale(param, layer_id):
            param.div_(math.sqrt(2.0 * layer_id))

        for layer_id, layer in enumerate(self.blocks):
            rescale(layer.attn.proj.weight.data, layer_id + 1)
            rescale(layer.mlp.fc2.weight.data, layer_id + 1)

    def _init_weights(self, m):
        if isinstance(m, nn.Linear):
            trunc_normal_(m.weight, std=.02)
            if isinstance(m, nn.Linear) and m.bias is not None:
                nn.init.constant_(m.bias, 0)
        elif isinstance(m, nn.LayerNorm):
            nn.init.constant_(m.bias, 0)
            nn.init.constant_(m.weight, 1.0)

    def init_weights(self, pretrained=None):
        """Initialize the weights in backbone.
        Args:
            pretrained (str, optional): Path to pre-trained weights.
                Defaults to None.
        """
        logger = get_root_logger()
        logger.info("test")

        def _init_weights(m):
            if isinstance(m, nn.Linear):
                trunc_normal_(m.weight, std=.02)
                if isinstance(m, nn.Linear) and m.bias is not None:
                    nn.init.constant_(m.bias, 0)
            elif isinstance(m, nn.LayerNorm):
                nn.init.constant_(m.bias, 0)
                nn.init.constant_(m.weight, 1.0)
        
        if (isinstance(self.init_cfg, dict) and self.init_cfg.get('type') == 'Pretrained'):
            self.apply(_init_weights)
            logger = get_root_logger()
            logger.info(f'load checkpoint from {self.init_cfg["checkpoint"]}')
            load_checkpoint(self, self.init_cfg['checkpoint'], strict=False, logger=logger)

        elif pretrained is None:
            self.apply(_init_weights)
        else:
            raise TypeError('pretrained must be a str or None')
```
